src/hooks/useQueryParametersState.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { useSearchParams } from './useSearchParams';
+
+enum HistoryStrategy {
+  Push = 'push',
+  Replace = 'replace',
+}
+
+interface QueryParametersStateProps<ValueType> {
+  key: string;
+  initialValue: ValueType;
+  encode?: (value: ValueType) => string;
+  decode?: (value: string) => ValueType;
+  strategy?: HistoryStrategy;
+}
+
+const useQueryParametersState = <ValueType>({
+  key,
+  initialValue,
+  encode = JSON.stringify,
+  decode = JSON.parse,
+  strategy = HistoryStrategy.Push,
```

Comment:
I think this should be `HistoryStrategy.Replace` by default. I love that we have the option to put it in the push history but I think my expectation as a user would be if I had gone to the checks page, done some filtering then hit back I'd go back to the page I've come from.

I could easily be convinced the other way, though. I know often in Grafana I use browser back after using the time range filter and I love that functionality.

What do you think?

Reply:
Sure thing! I missed this when changing the implementation. I agree with you. Fixed!